File: services/project/internal/data/models/sprint.go
package models

import "time"

type Sprint struct {
	ID          string    `db:"id"`
	Title       string    `db:"title"`
	Description string    `db:"description"`
	StartDate   time.Time `db:"start_date"`
	EndDate     time.Time `db:"end_date"`
	ProjectID   string    `db:"project_id"`
	CreatedAt   time.Time `db:"created_at"`
	UpdatedAt   time.Time `db:"updated_at"`
}

File: services/frontend/src/lib/sprint/sprint.ts
import { Task } from "../task/task";
import { Pagination } from "../utils/list";

export type Sprint = {
  id: string;
  title: string;
  description: string;
  start_date: {
    seconds: number;
  };
  end_date: {
    seconds: number;
  };
  project_id: string;
  created_at: {
    seconds: number;
  };
  updated_at: {
    seconds: number;
  };
  tasks: Array<Task>;
};

export type SprintCreation = Omit<Sprint, "id" | "created_at" | "updated_at">;

export type SprintFilter = Pagination & {
  project_id?: string;
  title?: string;
};
